js/modules/form_input_textfield.js

```diff
@@ -1,122 +1,138 @@
-<?php
-header("Content-Type: application/javascript; charset=UTF-8", true);
+let className = "Catalyst\\Form\\Field\\TextField";
+window.log("Form input handlers", "Registering "+className);
 
-define("ROOTDIR", "../../");
-define("REAL_ROOTDIR", "../../");
+if (!window.hasOwnProperty("formInputHandlers")) {
+	window.formInputHandlers = {};
+}
 
-define("NO_SESSION", true);
+let VALID = 0;
+let INVALID = 1;
+let MISSING = 2;
 
-require_once REAL_ROOTDIR."src/initializer.php";
-use \Catalyst\Form\Field\TextField;
-?>
-{
-	<?php require_once REAL_ROOTDIR."js/modules/form_input_header.js"; ?>
-	
-	const className = <?= json_encode(TextField::class) ?>;
+class TextField extends HTMLElement {
+	constructor(properties) {
```

Comment:
`properties` should have a default value, since it's unsafely accessed everywhere, and the undefined check below assigns the parsed output of an attribute, which can also be undefined